src/main/java/apoc/merge/Merge.java

```diff
@@ -25,26 +25,49 @@
         String labels = labelNames.stream().map(s -> wrapInBacktics(s)).collect(Collectors.joining(":"));
 
         Map<String, Object> params = new HashMap<>();
-        params.put("props", Util.merge(identProps, props));
+        params.put("identProps", identProps);
+        params.put("props", props);
         String identPropsString = buildIdentPropsString(identProps);
 
         final String cypher = "MERGE (n:" + labels + "{" + identPropsString + "}) ON CREATE SET n += $props RETURN n";
         Node node = Iterators.single(db.execute(cypher, params ).columnAs("n"));
         return Stream.of(new NodeResult(node));
     }
 
-    private String wrapInBacktics(String s) {
-        return "`" + s + "`";
+    @Procedure(value="apoc.merge.node.eager", mode = Mode.WRITE, eager = true)
+    @Description("apoc.merge.node.eager(['Label'], identProps:{key:value, ...}, config:{onCreateProps:{key:value,...}, onMatchProps:{key:value,...}}) - merge node with dynamic labels, with support for setting properties ON CREATE or ON MATCH")
+    public Stream<NodeResult> nodeEager(@Name("label") List<String> labelNames, @Name("identProps") Map<String, Object> identProps, @Name(value="config", defaultValue = "{}") Map<String, Object> config) {
+        if ((identProps==null) || (identProps.isEmpty())) {
+            throw new IllegalArgumentException("you need to supply at least one identifying property for a merge");
+        }
+
+        validateKeys(config.keySet(), Lists.newArrayList("onCreateProps", "onMatchProps"));
+
+        String labels = labelNames.stream().map(s -> wrapInBacktics(s)).collect(Collectors.joining(":"));
```

Comment:
there is Util.quote which does a better job of this.

Reply:
In place of the wrapInBacktics() call? Okay, will replace.